pkg/segment/writer/stats/segstats.go

```diff
@@ -36,7 +36,7 @@ func AddSegStatsNums(segstats map[string]*SegStats, cname string,
 	var stats *SegStats
 	var ok bool
 	stats, ok = segstats[cname]
-	if !ok {
+	if !ok || !stats.IsNumeric {
 		numStats := &NumericStats{
```

Comment:
I don't quite understand why we need this. Can you explain it to me? I think in your PR description you mentioned this was to prioritize numeric stats over string stats?

Reply:
Consider a filter like `app_name=Bracecould` there is 1 column value having numeric value say (301952649), 3 column values having strings (with numeric values) and 2 backfilled values. When a segstats query is fired `app_name=Bracecould | stats avg(variable_col_4), max(variable_col_18), sum(variable_col_17)`, the num stats are calculated for average and incoming string stats are ignored which results in the avg being a single value i.e. `301952649`. When the same kind of query but in form of an RRCCmd like `app_name=Bracecould | eval n = 1 | stats avg(variable_col_4), max(variable_col_18), sum(variable_col_17)` is fired, the count value in numeric stats is changed differently meaning a numeric stat got initialized and then a string value came in and incremented the count resulting in average being different. Due to this inconsistency, I was planning to override the values in case if a numeric value comes in and stringStats are present, vice-versa we would ignore if a a numeric stats is present and string comes in.

Reply:
Moreover, if the stats is already initialized and we call `processStats` method we are panicking because NumStats are not initialized.